### DISCUSSION TITLE
Schedule refreshof dataset in Power BI when source is TM1py

### DISCUSSION THREAD

**Discussion by mikeponeill1978** [cubewise-code/tm1py#744] [General]
**Describe what did you try to do with TM1py**
I have created a Power BI dashboard where the data source is TM1 via a python scrip using TM1py functions and am happy with the results. 

I now want to schedule a refresh of the dataset in Power BI. However when I try to do so I get the message:

"You can't schedule refresh of this dataset because the following data sources currentl don't support refresh:"

Does anyone now a way around this?
Cheers

**Comment by MariusWirtz** [comment]
I think you need to use a Personal Gateway in PowerBI. Like in this article:
https://docs.microsoft.com/en-us/power-bi/connect-data/service-gateway-personal-mode

Please let us know about your experiences. I haven't implemented this personally yet.

**Comment by zsoltmoravcsik** [comment]
It is working. but You need to install personal gateway on the server,
schedule a task then in power no you can trigger the personal gateway
refresh.
Known issue: this not enterprise ready yet base din Microsoft docs
